src/main/java/graph/UserGraph.java

```diff
@@ -1,248 +1,31 @@
 package graph;
 
-
 import graph.exceptions.ObelixNodeNotFoundException;
-import org.neo4j.graphdb.DynamicLabel;
-import org.neo4j.graphdb.GraphDatabaseService;
-import org.neo4j.graphdb.Label;
-import org.neo4j.graphdb.Node;
-import org.neo4j.graphdb.NotFoundException;
-import org.neo4j.graphdb.Path;
-import org.neo4j.graphdb.Relationship;
-import org.neo4j.graphdb.Transaction;
-import org.neo4j.graphdb.traversal.Evaluators;
-import org.neo4j.graphdb.traversal.Uniqueness;
+import graph.interfaces.GraphDatabase;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.HashMap;
-import java.util.LinkedList;
 import java.util.List;
-import java.util.Map;
-
-import static events.NeoHelpers.getAllNodes;
-import static events.NeoHelpers.getUserNode;
-import static events.NeoHelpers.makeSureTheUserDoesNotExceedMaxRelationshipsLimit;
 
 public class UserGraph {
```

Comment:
I agree that we can remove UserGraph, but we still need user objects. Maybe we should create a:

interface GraphNode

class User implements GraphNode

and

class Item implements GraphNode

?